protostar/protostar_cli.py

```diff
@@ -24,7 +25,7 @@ def _apply_pythonpath():
         sys.path.extend(split_paths)
 
 
-class ProtostarCLI(CLIApp):
+class ProtostarCLI(CLIApp, CommandNamesProvider):
```

Comment:
I think it would be better to keep provider as a property not as a protocol. It is very hard to reason about.

Reply:
`CommandNamesProvider` property would have to operate on internals of `ProtostarCLI` which breaks the encapsulation.

> It is hard to reason about.

This is a delegate pattern (I need to change `CommandNamesProvider` to `CommandNamesProviderDelegate`)
https://dilloncodes.com/delegate-pattern-in-swift

Reply:
I think new name is worse, Delegate pattern is something that not every python programmer is familliar with